src/solvers/flattening/bv_dimacs.cpp

```diff
@@ -49,21 +49,24 @@ bool bv_dimacst::write_dimacs(std::ostream &out)
   // dump mapping for selected bit-vectors
   for(const auto &m : get_map().mapping)
   {
-    const boolbv_mapt::literal_mapt &literal_map = m.second.literal_map;
+    const auto &literal_map = m.second.literal_map;
 
     if(literal_map.empty())
       continue;
 
     out << "c " << m.first;
 
     for(const auto &lit : literal_map)
-      if(!lit.is_set)
-        out << " "
-            << "?";
-      else if(lit.l.is_constant())
-        out << " " << (lit.l.is_true() ? "TRUE" : "FALSE");
+    {
+      out << ' ';
+
+      if(!m.second.is_set)
```

Comment:
Same thing here, I guess. With the changes to me it looks like now the case where the whole map is not set is "separate" logically.

Reply:
I first wanted to keep the changes to a minimum, but ultimately that wasn't possible anyway. I have now clarified the code structure.